Framework/API/test/AlgorithmTest.h

```diff
@@ -787,6 +806,86 @@ class AlgorithmTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(algCastConst, algCastNonConst);
   }
 
+  void testIndexingAlgorithm_declareIndexPropertyMethod() {
+    IndexingAlgorithm indexAlg;
+    TS_ASSERT_THROWS_NOTHING(
+        indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace"));
```

Comment:
I was surprised here. Is `declareIndexProperty` public? Is `declareProperty` public??